[WebGPUSwift] move internal headers to WebGPU_Internal module

nmahendru opened the pull request:
#### 1357bd02984ffa9c99bdb69606af675ae3b7a8e9
<pre>
[WebGPUSwift] move internal headers to WebGPU_Internal module
<a href="https://bugs.webkit.org/show_bug.cgi?id=283544">https://bugs.webkit.org/show_bug.cgi?id=283544</a>
<a href="https://rdar.apple.com/140394391">rdar://140394391</a>

Reviewed by Mike Wyrzykowski and Elliott Williams.

Currently there is a long list of headers at the bottom of WebGPUExt.h that
were added to let them be imported into Swift via WebGPU.modulemap.
Although they are hidden behing the flag __WEBGPU__, the headers still live in
WebGPU.modulemap which is wrong philosophically as these headers are only to be
consumed by Swift in WebGPU.
Now, since we have another internal module i.e. WebGPU_Internal, it&apos;s cleaner
to move them there.
This should not cause any change in functionality.

* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebGPU/WebGPU/Adapter.h:
* Source/WebGPU/WebGPU/Buffer.h:
* Source/WebGPU/WebGPU/CommandEncoder.h:
* Source/WebGPU/WebGPU/Device.h:
* Source/WebGPU/WebGPU/HardwareCapabilities.h:
* Source/WebGPU/WebGPU/Internal/WebGPUSwiftInternal.h:
* Source/WebGPU/WebGPU/QuerySet.h:
* Source/WebGPU/WebGPU/Texture.h:
* Source/WebGPU/WebGPU/WebGPUExt.h:

Canonical link: <a href="https://commits.webkit.org/287367@main">https://commits.webkit.org/287367@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/349959b2e2c3029e839ae9f1d23785a50382ba57

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/79220 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/58248 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/32592 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/83842 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/30389 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/67340 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/6518 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/61972 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19871 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/82287 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/52026 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/72187 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42277 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/49372 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/28775 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/70484 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/26764 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/85237 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/6518 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/4514 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/70216 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/6681 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/68059 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69472 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13518 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/12341 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12253 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/6475 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/6399 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/9866 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8191 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple reviewed (commented):
Do you really need all of these? 

```
#include "WebGPU.h"
#include "WebGPUExt.h"
```

They are both included from `config.h`, so I would have thought that they are already imported at the top of every translation unit.

emw-apple reviewed (approved):
r+, but please remove the extra includes if you're able to!

nmahendru committed:
[WebGPUSwift] move internal headers to WebGPU_Internal module
https://bugs.webkit.org/show_bug.cgi?id=283544
rdar://140394391

Reviewed by Mike Wyrzykowski and Elliott Williams.

Currently there is a long list of headers at the bottom of WebGPUExt.h that
were added to let them be imported into Swift via WebGPU.modulemap.
Although they are hidden behing the flag __WEBGPU__, the headers still live in
WebGPU.modulemap which is wrong philosophically as these headers are only to be
consumed by Swift in WebGPU.
Now, since we have another internal module i.e. WebGPU_Internal, it's cleaner
to move them there.
This should not cause any change in functionality.

* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebGPU/WebGPU/Adapter.h:
* Source/WebGPU/WebGPU/Buffer.h:
* Source/WebGPU/WebGPU/CommandEncoder.h:
* Source/WebGPU/WebGPU/Device.h:
* Source/WebGPU/WebGPU/HardwareCapabilities.h:
* Source/WebGPU/WebGPU/Internal/WebGPUSwiftInternal.h:
* Source/WebGPU/WebGPU/QuerySet.h:
* Source/WebGPU/WebGPU/Texture.h:
* Source/WebGPU/WebGPU/WebGPUExt.h:

Canonical link: https://commits.webkit.org/287367@main
